packages/cli-repl/test/repl-helpers.ts

```diff
@@ -77,29 +77,40 @@ async function readReplLogfile(logPath: string) {
 }
 
 
-const fakeExternalEditor = async(output?: string) => {
-  const base = path.resolve(__dirname, '..', '..', '..', 'tmp', 'test', `${Date.now()}`, `${Math.random()}`);
-  const tmpDoc = path.join(base, 'editor-script.js');
+const fakeExternalEditor = async(
+  { output, expectedExtension, tmpdir }: { output?: string, expectedExtension?: string, tmpdir: string }
+) => {
+  const editor = (expectedExtension === '.mongodb') ? 'code' : 'editor';
+  const tmpDoc = path.join(tmpdir, editor);
   let script: string;
 
   if (typeof output === 'string') {
-    script = `(async () => {
+    script = `#!/usr/bin/env node
+    (async () => {
       const tmpDoc = process.argv[process.argv.length - 1];
       const { promises: { writeFile } } = require("fs");
+      const assert = require("assert");
+      const path = require("path");
+
       await writeFile(tmpDoc, ${JSON.stringify(output)}, { mode: 0o600 });
-    })()`;
+
+      if (${JSON.stringify(expectedExtension ?? '')}) {
+        assert.strictEqual(path.extname(tmpDoc), ${JSON.stringify(expectedExtension)});
+      }
```

Comment:
- Can we do this before the `writeFile()` call so that writing to the doc doesn’t accidentally make a test pass at some point?
- Can we also similarly verify that the `--wait` flag was actually passed to the script?